### DISCUSSION TITLE
Unflashing crazyflie

### DISCUSSION THREAD

**Discussion by Pandagineer** [bitcraze/discussions#231] [Q&A] [answered]
Hello. I have flashed (after making) the app_hello_world project. I successfully see "Hello World" in the CFclient Console appear every 2 seconds. Now I would like to "unflash" this. How can I do this? Thanks.

**Comment by knmcguire** [comment, marked as answer]
Hi!

In this case you should just flash the regular non-app version of the firmware. Or to flash the latest release again through the cfclient: https://www.bitcraze.io/documentation/repository/crazyflie-clients-python/master/userguides/userguide_client/#firmware-upgrade

**Reply by Pandagineer** [reply]
Ok. But just to make sure I understand: if I have flashed it with 20 different programs, but one of them is causing a problem, I have to flash the non-app version, and then re-flash my 19 good programs. Do I understand correctly?

**Reply by knmcguire** [reply]
Ah no, I think there is a bit of misunderstanding on that point, as it is not adding the program like on an flash drive. `Make cload` will actually remove the old code completely and flash it with the one you have build. 

So you can't flash it with 20 different programs, since only the last one remains.

This is all within how we have set it all up in terms of build tools of course. In reality, you could flash different programs on different addresses on the memory of the STM, but that requires advanced embedded programming knowledge, so you shouldn't worry about that if you are just beginning (and with the current status, that will brick/break the crazyflie)

For now, just assume every time you do 'make cload' there is only one (autopilot) program on the crazyflie at all times, namely the latest you build.

**Reply by Pandagineer** [reply]
Also, I noticed something. The "hello world" message has stopped appearing. Here are a few things I have done recently:
- the crazyflie battery has died, and I have recharged
- I have closed the CFclient, and later reopened
- I have flashed it with another example (app_internal_param_log), which is properly echoing yaw and estimator information
- I have ran a few pythong clib examples

Given the above list, does it make sense that I no longer see "hello word" in the console?

Thanks!

**Reply by Pandagineer** [reply]
(I just read your reply, and it also answers my most recent question. I understand why I don't see "hello word", since I have flashed it with something else. This discussion can be closed. Thanks again.)

**Reply by knmcguire** [reply]
no worries :) I've marked the solution as an answer. If you need any more help with something else you open up another discussions 😄
